network/src/protocols/identify.rs

```diff
@@ -109,3 +152,102 @@ impl Callback for IdentifyCallback {
         MisbehaveResult::Disconnect
     }
 }
+
+#[derive(Clone)]
+struct Identify {
+    name: String,
+    flags: Flags,
+    encode_data: Vec<u8>,
+}
+
+impl Identify {
+    fn new(name: String, flags: Flags) -> Self {
+        Identify {
+            name,
+            flags,
+            encode_data: Vec::default(),
+        }
+    }
+
+    fn encode(&mut self) -> &[u8] {
+        if self.encode_data.is_empty() {
+            let mut fbb = FlatBufferBuilder::new();
+            let name = fbb.create_string(&self.name);
+
+            let mut builder = FbsIdentifyBuilder::new(&mut fbb);
+
+            builder.add_flag(self.flags.0);
+            builder.add_name(name);
+
+            let data = builder.finish();
+            fbb.finish(data, None);
+            self.encode_data = fbb.finished_data().to_vec();
+        }
+
+        &self.encode_data
+    }
+
+    fn verify(&self, data: &[u8]) -> Option<Flags> {
+        let fbs_message = get_root::<FbsIdentify>(data).ok()?;
+
+        match (fbs_message.name(), fbs_message.flag()) {
+            (Some(raw_name), flag) => {
+                if self.name != raw_name {
+                    debug!(
+                        "Not the same chain, self: {}, remote: {}",
+                        self.name, raw_name
+                    );
+                    return None;
+                }
+
+                if flag == 0 {
```

Comment:
What does `flag == 0` mean? Maybe add a comment?

Reply:
```
#[repr(u64)]
enum Flag {
    /// Support all protocol
    FullNode = 0x1,
}
```
zero is flatbuffer default value